### ISSUE TITLE
404 not found (installation on mac)

### ISSUE THREAD

**Issue by jasond88** [Sonerezh/sonerezh#123]
Hey guys,

I'm looking forward to getting this thing to work, I demoed it and it's awesome. Unfortunately I'm stumped, and spent hours and still haven't gotten it able to work.

My apache server is running, got mysql running and I believe all the tables are there. I enabled mod_rewrite and double checked all the other recommended settings in the installation documentation.

Here's a few specifics:
my path to the http main dir is /Users/(username)/Sites
I set my server on port 6060 instead of 80 because my ISP blocks 80.

I go to localhost:6060, get directed to localhost:6060/install and receive this error on the page:
"Not Found
The requested URL /install was not found on this server."

Any ideas what I did wrong?

Thanks

**Comment by nodiscc** [comment]
What do your apache (access, error) logs say?

**Comment by jasond88** [comment]
my error log just keeps repeating:
[Thu Oct 08 22:28:17.445710 2015] [core:notice] [pid 33879] AH00094: Command line: '/usr/sbin/httpd -D FOREGROUND'
[Thu Oct 08 22:47:30.989495 2015] [mpm_prefork:notice] [pid 33879] AH00169: caught SIGTERM, shutting down
[Thu Oct 08 22:47:31.277439 2015] [mpm_prefork:notice] [pid 34242] AH00163: Apache/2.4.16 (Unix) PHP/5.5.27 configured -- resuming normal operations

I think this is because of my apache restarts.

My access log:

(I x'ed the IP from the log)
xx.xx.xx.xx - - [08/Oct/2015:16:45:12 -0400] "GET /install HTTP/1.1" 404 205
xx.xx.xx.xx - - [08/Oct/2015:16:45:18 -0400] "GET / HTTP/1.1" 500 2533
xx.xx.xx.xx - - [08/Oct/2015:16:45:18 -0400] "GET /css/bootstrap.min.css HTTP/1.1" 404 219
xx.xx.xx.xx - - [08/Oct/2015:16:45:18 -0400] "GET /css/jquery.fs.selecter.min.css HTTP/1.1" 404 228
xx.xx.xx.xx - - [08/Oct/2015:16:45:18 -0400] "GET /css/pace.css HTTP/1.1" 404 210
xx.xx.xx.xx - - [08/Oct/2015:16:45:18 -0400] "GET /css/notify.css HTTP/1.1" 404 212
xx.xx.xx.xx - - [08/Oct/2015:16:45:18 -0400] "GET /css/slider.css HTTP/1.1" 404 212
xx.xx.xx.xx - - [08/Oct/2015:16:45:19 -0400] "GET /css/style.css HTTP/1.1" 404 211
xx.xx.xx.xx - - [08/Oct/2015:16:45:20 -0400] "GET /js/lazyload.min.js HTTP/1.1" 404 216
xx.xx.xx.xx - - [08/Oct/2015:16:45:26 -0400] "GET /install HTTP/1.1" 404 205

**Comment by jasond88** [comment]
I made a bit of progress, but still unable to get to the installation screen. I now get a blank page when I go to the server home.

Let me report what I did:
I made a really stupid mistake in httpd.conf and accidentally changed ServerRoot to my server directory, which is not right, and supposed to be "/usr" (at least for Mac it is).

When I go to my error log now, at least I'm getting something that makes sense:

PHP Fatal error:  Uncaught exception 'CacheException' with message 'Cache engine "_cake_core_" is not properly configured. Ensure required extensions are installed, and credentials/permissions are correct' in /Users/(myusername)/Sites/lib/Cake/Cache/Cache.php:186\nStack trace:\n#0 /Users/jasondourgarian/Sites/lib/Cake/Cache/Cache.php(151): Cache::_buildEngine('_cake_core_')\n#1 /Users/(myusername)/Sites/app/Config/core.php(379): Cache::config('_cake_core_', Array)\n#2 /Users/(myusername)/Sites/lib/Cake/Core/Configure.php(72): include('/Users/(myusername)...')\n#3 /Users/(myusername)/Sites/lib/Cake/bootstrap.php(175): Configure::bootstrap(true)\n#4 /Users/(myusername)/Sites/app/webroot/index.php(94): include('/Users/(myusername)...')\n#5 /Users/(myusername)/Sites/index.php(41): require('/Users/(myusername)...')\n#6 {main}\n  thrown in /Users/(myusername)/Sites/lib/Cake/Cache/Cache.php on line 186

**Comment by jasond88** [comment]
HA! Finally figured it out!
I thought I had all the permissions and settings configured properly but I didn't!
Apache, php, and mysql were all configured properly.
Later I realized that it worked in a subfolder but still the 404 error in the root directory.

It is imperative that the owner and group of the files/folders are the same as the user and group set in the apache. Also, it is very important to have a .htaccess file configured properly. (I ended up not having a .htaccess present)

Here is what the default .htaccess file is (if you don't have this make sure it exists in your main directory if you have the same problem)

<IfModule mod_rewrite.c>
   RewriteEngine on
   RewriteRule    ^$ app/webroot/    [L]
   RewriteRule    (.*) app/webroot/$1 [L]
</IfModule>

Anyway, I hope this helps someone that was in the same boat as me.

Cheers!
